documentation/benchmarks/graphical_library_benchmark/graphical_library_benchmark.md

```diff
@@ -0,0 +1,113 @@
+| Memory Usage (MB)   | 1.5     | 1        | 1.5     |
+| Frame time (avg MS) | 23-25   | 1-2      | 1-2     |
+
+
+### 2. **Test Case 2: Scrolling Background**
+
+This test measures performance while rendering a smooth scrolling background (parallax or tile-based scrolling).
+
+| **Metric**              | **SFML** | **SDL2** | **Allegro** |
+|-------------------------|----------|----------|-------------|
+| CPU Utilization (%)     | 22       | 18       | 4           |
+| Memory Usage (MB)       | 1.5      | 1.7      | 1.5         |
+| Frame time (avg MS)     | 2-5      | 2-3      | 500-7000    |
+
+## Summary of Findings
+
+### Performance Comparison
+
+- **Rendering 1000 Sprites:**
+  - SFML shows the lowest CPU utilization and moderate frame time, making it efficient for rendering numerous sprites.
+  - SDL2 has higher CPU usage but manages to have excellent frame times, reflecting its choice between performance and resource consumption.
+  - Allegro demonstrates excellent frame times and really low CPU usage, but with comparable memory usage to SFML.
+- **Scrolling Background:** 
+  - SFML and SDL2 have similar frame times and are capable of handling scrolling backgrounds smoothly.
+  - Allegro exhibits a significant range in frame time, which can be attributed to its rendering method. While it has the lowest CPU usage, its performance is less consistent.
+### Pros and Cons
+
+| **Library**   | **Pros**                                   | **Cons**                                   |
+|---------------|--------------------------------------------|--------------------------------------------|
+| **SFML**      | Easy to use, good documentation            | Less optimized for low-end hardware        |
+| **SDL2**      | Highly portable, strong community support  | API complexity for beginners               |
+| **Allegro**   | Mature, great performance in some cases    | Less active development than others        |
+
+---
+
+## Conclusion
+
+**Based on the benchmark results,**
+
+SDL2 is recommended for 2D game engine development. It balances performance and resource usage effectively across both tests that have been done.
+
+Additionally, SDL2’s strong community support and availability of external libraries, such as those for complex particle systems, make it a compelling choice for more advanced features and customizations.
+
+SFML is also a strong contender, particularly for projects requiring ease of use and simpler setup.
+
+However, Allegro, despite its efficient performance in some areas, shows less consistent results in scrolling scenarios and is less actively developed compared to SDL2.
+
+#### In conclusion, **SDL2**'s performance and the ecosystem of external libraries make it the preferred choice for developing a robust 2D game engine with complex features as we need it.
+
+---
+
+## Appendix
+
+### Test Scripts
+
+Include the test scripts and commands used to run the benchmarks.
```

Comment:
You can add a note to remind people that they should download each library before compiling

Reply:
I've added the commands i used to install the librairies